Sources/GtkBackend/GtkBackend.swift

```diff
@@ -954,22 +954,56 @@ public final class GtkBackend: AppBackend {
         gtk_native_dialog_show(chooser.gobjectPointer.cast())
     }
 
-    public func createTapGestureTarget(wrapping child: Widget) -> Widget {
-        let gesture = Gtk.GestureClick()
-        child.addEventController(gesture)
+    public func createTapGestureTarget(wrapping child: Widget, gesture: TapGesture) -> Widget {
+        var gtkGesture: GestureSingle
+        switch gesture.kind {
+            case .primary:
+                gtkGesture = GestureClick()
+            case .secondary:
+                gtkGesture = GestureClick()
+                gtk_gesture_single_set_button(gtkGesture.opaquePointer, 3)
+            case .longPress:
+                gtkGesture = GestureLongPress()
+        }
+        child.addEventController(gtkGesture)
         return child
     }
 
     public func updateTapGestureTarget(
         _ tapGestureTarget: Widget,
+        gesture: TapGesture,
         action: @escaping () -> Void
     ) {
-        let gesture = tapGestureTarget.eventControllers[0] as! Gtk.GestureClick
-        gesture.pressed = { _, nPress, _, _ in
-            guard nPress == 1 else {
-                return
-            }
-            action()
+        switch gesture.kind {
+            case .primary:
+                let gesture =
+                    tapGestureTarget.eventControllers.first {
+                        $0 is GestureClick && gtk_gesture_single_get_button($0.opaquePointer) == 1
```

Comment:
Use `GDK_BUTTON_PRIMARY` and `GDK_BUTTON_SECONDARY` instead of `1` and `3` for primary and secondary clicks.

Reply:
Oh, wait, G**D**K. Those constants are found but they're the wrong type (`Int32` vs `UInt32`).

Reply:
Should be able to just convert the type, I've confirmed that they're the right values. Use `guint(...)` to convert it in case they ever change the underlying type or it's different on some strange system etc.